### PR TITLE
Adds CreateRelease Operation

### PR DESCRIPTION
**Pull request by rain-on**
This includes the Release Api Classes:
* ReleaseApi
* ReleaseResource
* ReleaseResourceWithLinks
* ReleasePaginatedCollection

### CODE REVIEW

File: octopus-sdk/src/main/java/com/octopus/sdk/model/commands/CreateReleaseCommandParameters.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  @SerializedName("packagePrerelease")
+  private String packagePrerelease;
+
+  @SerializedName("whatIf")
+  private boolean whatIf;
+
+  @SerializedName("deployToEnvironmentIdsOrNames")
+  private List<String> deployToEnvironmentIdsOrNames;
+
+  public CreateReleaseCommandParameters(final String projectIdOrName, final String packageVersion) {
```
Comment (long-sean) [inline octopus-sdk/src/main/java/com/octopus/sdk/model/commands/CreateReleaseCommandParameters.java:70]: Preconditions?

File: octopus-sdk/src/main/java/com/octopus/sdk/model/release/ReleaseResource.java [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+import com.octopus.sdk.model.BaseResource;
+
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import com.google.gson.annotations.SerializedName;
+
+/** ReleaseResource */
```
Comment (long-sean) [inline octopus-sdk/src/main/java/com/octopus/sdk/model/release/ReleaseResource.java:30]: Unneeded commentary?

File: octopus-sdk/src/main/java/com/octopus/sdk/model/commands/CreateReleaseCommandParameters.java (hunk 1 above)
Comment (rain-on) [reply octopus-sdk/src/main/java/com/octopus/sdk/model/commands/CreateReleaseCommandParameters.java:70]: added

File: octopus-sdk/src/main/java/com/octopus/sdk/model/release/ReleaseResource.java (hunk 2 above)
Comment (rain-on) [reply octopus-sdk/src/main/java/com/octopus/sdk/model/release/ReleaseResource.java:30]: removed

**Review by long-sean — APPROVED** [review]
Good to go

**Pull request merged**